packages/context/src/__tests__/unit/binding.unit.ts

```diff
@@ -214,6 +214,56 @@ describe('Binding', () => {
     });
   });
 
+  describe('cache', () => {
+    let spy: SinonSpy;
+    beforeEach(() => {
+      spy = sinon.spy();
+    });
+
+    it('clears cache if scope changes', () => {
+      const indexBinding = ctx
+        .bind<number>('index')
+        .toDynamicValue(spy)
+        .inScope(BindingScope.SINGLETON);
+      ctx.getSync(indexBinding.key);
+      sinon.assert.calledOnce(spy);
+      // Singleton
+      ctx.getSync(indexBinding.key);
+      sinon.assert.calledOnce(spy);
+      indexBinding.inScope(BindingScope.CONTEXT);
+      ctx.getSync(indexBinding.key);
+      sinon.assert.calledTwice(spy);
```

Comment:
The test is easier to understand now. I am proposing to improve it even more and use [`spy.resetHistory()`](https://sinonjs.org/releases/v7.3.1/spies/#spyresethistory) in between context calls.

```ts
ctx.getSync(indexBinding.key);
sinon.assert.calledOnce(spy);

// Singleton is caching the value
spy.resetHistory();
ctx.getSync(indexBinding.key);
sinon.assert.notCalled(spy);

// Scope change clears the cache
indexBinding.inScope(BindingScope.CONTEXT);
spy.resetHistory();
ctx.getSync(indexBinding.key);
sinon.assert.calledOnce(spy);
```